observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetryDeploymentProcessor.java

```diff
@@ -24,13 +21,15 @@
 import org.wildfly.extension.opentelemetry.api.OpenTelemetryCdiExtension;
 import org.wildfly.extension.opentelemetry.api.WildFlyOpenTelemetryConfig;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
 class OpenTelemetryDeploymentProcessor implements DeploymentUnitProcessor {
-    private final boolean useServerConfig;
-    private final WildFlyOpenTelemetryConfig serverConfig;
+    private final AtomicReference<WildFlyOpenTelemetryConfig> configSupplier;
```

Comment:
Nit: As this DUP only reads this reference, better to use a Supplier<WildFlyOpenTelemetryConfig>.
e.g.
```java
AtomicReference<WildFlyOpenTelemetryConfig> configReference = new AtomicReference<>();
// ...
target.addDeploymentProcessor(..., new OpenTelemetryDeploymentProcessor(configReference::get));